plugin/src/main/java/org/opensearch/ml/utils/RestActionUtils.java

```diff
@@ -87,13 +96,24 @@ public static FetchSourceContext getSourceContext(RestRequest request, SearchSou
         } else {
             // When user does not set the _source field in search model api request, searchSourceBuilder.fetchSource becomes null
             if (!userAgent.contains(OPENSEARCH_DASHBOARDS_USER_AGENT)) {
-                return new FetchSourceContext(true, Strings.EMPTY_ARRAY, UI_METADATA_EXCLUDE);
+                return new FetchSourceContext(true, Strings.EMPTY_ARRAY, ArrayUtils.add(UI_METADATA_EXCLUDE, "model_content"));
             } else {
-                return null;
+                return new FetchSourceContext(true, Strings.EMPTY_ARRAY, new String[] { "model_content" });
             }
         }
     }
 
+    /**
+     * Return FetchSourceContext
+     * @param returnModelContent if the model content should be returned
+     */
+    public static FetchSourceContext getFetchSourceContext(boolean returnModelContent) {
+        if (!returnModelContent) {
+            return new FetchSourceContext(true, Strings.EMPTY_ARRAY, new String[] { "content", "model_content" });
+        }
+        return new FetchSourceContext(true, Strings.EMPTY_ARRAY, Strings.EMPTY_ARRAY);
```

Comment:
Will it return all fields by setting includes and excludes as empty array?

Reply:
Yes, it returns everything if returnModelContent=true by using Strings.EMPTY_ARRAY in both includes and excludes. Tested it with several models. Check this https://www.elastic.co/guide/en/elasticsearch/client/java-rest/current/java-rest-high-document-get.html.